platforms/posix/cmake/sitl_target.cmake

```diff
@@ -21,15 +21,37 @@ px4_add_git_submodule(TARGET git_jmavsim PATH "${PX4_SOURCE_DIR}/Tools/jMAVSim")
 
 # Add support for external project building
 include(ExternalProject)
-include(ProcessorCount)
 
-set(build_cores 1)
+# Estimate an appropriate number of parallel jobs
+cmake_host_system_information(RESULT AVAILABLE_PHYSICAL_MEMORY QUERY AVAILABLE_PHYSICAL_MEMORY)
+cmake_host_system_information(RESULT NUMBER_OF_LOGICAL_CORES QUERY NUMBER_OF_LOGICAL_CORES)
 
-ProcessorCount(N)
-if(N GREATER_EQUAL 4)
-	math(EXPR build_cores "${N} - 2")
+set(parallel_jobs 1)
+
+if(NOT NUMBER_OF_LOGICAL_CORES)
+	include(ProcessorCount)
+	ProcessorCount(NUMBER_OF_LOGICAL_CORES)
+endif()
+
+if(NOT AVAILABLE_PHYSICAL_MEMORY AND NUMBER_OF_LOGICAL_CORES GREATER_EQUAL 4)
+	# Memory estimate unavailable, use N-2 jobs
+	math(EXPR parallel_jobs "${NUMBER_OF_LOGICAL_CORES} - 2")
+endif()
+
+if (AVAILABLE_PHYSICAL_MEMORY)
+	# Allow an additional job for every 1.5GB of available physical memory
+	math(EXPR parallel_jobs "${AVAILABLE_PHYSICAL_MEMORY}/(3*1024/2)")
+else()
+	set(AVAILABLE_PHYSICAL_MEMORY "?")
+endif()
+
+if(parallel_jobs GREATER NUMBER_OF_LOGICAL_CORES)
+	set(parallel_jobs ${NUMBER_OF_LOGICAL_CORES})
 endif()
 
+message(STATUS  "${NUMBER_OF_LOGICAL_CORES} logical cores detected and ${AVAILABLE_PHYSICAL_MEMORY} megabytes of memory available.
```

Comment:
I would make this a debug message or get rid of it. Since it only applies to the simulation builds I think it's misleading.

Reply:
Changed to DEBUG and specified it's related to gazebo and ignition